apps/core/serializers.py

```diff
@@ -165,3 +165,43 @@ def validate(self, attrs):
             )
 
         return attrs
+
+
+class PasswordChangeSerializer(serializers.ModelSerializer):
+    password = serializers.CharField(
+        write_only=True,
+        required=True,
+        validators=[validate_password],
+    )
+    password2 = serializers.CharField(
+        write_only=True,
+        required=True,
+        validators=[validate_password],
+    )
+    old_password = serializers.CharField(
```

Comment:
It looks like we can use one serialiser for password reset and password change cases
If we make this field optional.

This will help us to reduce code duplication

Reply:
I don't think that this is a good idea, because:
If we will use both views (ResetChange and Change) with the same serializer (where the `old_password` field is optional) but different permissions of views, there will be more hardcoded elements in both views to check the `old_password` field, like:
```python
old_password = attrs.get("old_password")
if not old_password:
    attrs["old_password"] = None
    return attrs
``` 
Because we must end validation successfully, and if we will do that without `old_password`, there will be a possibility to change password without the old one.
Maybe i'm wrong but i see this case like that. Let's discuss it.

Reply:
The main idea is to reduce the amount of unnecessary code

```
class PasswordFormSerializer(serializers.Serializer):
    password = serializers.CharField(
        write_only=True,
        required=True,
        validators=[validate_password],
    )
    confirmed_password = serializers.CharField(
        write_only=True,
        required=True,
        validators=[validate_password],
    )

    def validate(self, attrs):
        if attrs["password"] != attrs["confirmed_password"]:
            raise serializers.ValidationError(
                {"password": "Password fields didn't match"}
            )

        return attrs


class PasswordChangeSerializer(PasswordFormSerializer):
    old_password = serializers.CharField(
        write_only=True,
        required=True,
        validators=[validate_password],
    )

    def validate(self, attrs):
        super().validate(attrs)

        user = self.context.get("user")

        if user.check_password(attrs["password"]):
            raise serializers.ValidationError(
                {"password": "The new password is the same as old one"}
            )
        if not user.check_password(attrs["old_password"]):
            raise serializers.ValidationError(
                {"old_password": "Existing password is incorrect"}
            )

        return attrs
```

Reply:
Yes, that's better. Didn't think that way -_-